.github/workflows/release-workflow.yml

```diff
@@ -0,0 +1,86 @@
+name: Release workflow
+# This workflow is triggered on creating tags to master or a opendistro release branch
+on:
+  push:
+  tags:
+    -'v*'
+    branches:
+      - master
+      - opendistro-*
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        java: [12]
+    # Job name
+    name: Build Alerting with JDK ${{ matrix.java }}
+    # This job runs on Linux
+    runs-on: ubuntu-latest
+    steps:
+      # This step uses the checkout Github action: https://github.com/actions/checkout
+      - name: Checkout Branch
+        uses: actions/checkout@v1
+      # This step uses the setup-java Github action: https://github.com/actions/setup-java
+      - name: Set Up JDK ${{ matrix.java }}
+        uses: actions/setup-java@v1
+        with:
+          java-version: ${{ matrix.java }}
+
+      # Building zip, deb and rpm files
+      - name: Build with Gradle
+        run: ./gradlew build buildDeb buildRpm --no-daemon -Dbuild.snapshot=false
+
+      - name: Create Artifact Path
+        run: |
+          mkdir -p alerting-artifacts
+          cp alerting/build/distributions/*.zip alerting-artifacts
+          cp alerting/build/distributions/*.deb alerting-artifacts
+          cp alerting/build/distributions/*.rpm alerting-artifacts
+          zip -r alerting.zip alerting-artifacts
+          echo ::set-env name=TAG_VERSION::${GITHUB_REF/refs\/tags\//}
+
+     # AWS authentication
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-west-2
+
+      # This step uses the upload-artifact Github action: https://github.com/actions/upload-artifact
+      - name: Upload Artifacts to S3
+        run: |
+          s3_path=s3://artifacts.opendistroforelasticsearch.amazon.com/downloads
+          aws s3 cp alerting-artifacts/*.zip $s3_path/elasticsearch-plugins/opendistro-alerting/
+          aws s3 cp alerting-artifacts/*.deb $s3_path/debs/opendistro-alerting/
+          aws s3 cp alerting-artifacts/*.rpm $s3_path/rpms/opendistro-alerting/
+          aws cloudfront create-invalidation --distribution-id ${{ secrets.DISTRIBUTION_ID }} --paths '/downloads/*'
+
+      - name: Create Github Draft Release
+        id: create_release
+        uses: actions/create-release@v1.0.0
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: ${{ github.ref }}
+          release_name: Release ${{ env.TAG_VERSION }}
+          draft: true
+          prerelease: false
+
+      - name: Upload Release Asset
+        id: upload-release-asset
+        uses: actions/upload-release-asset@v1.0.1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          upload_url: ${{ steps.create_release.outputs.upload_url }}
+          asset_name: alerting.zip
+          asset_path: alerting.zip
```

Comment:
This alerting.zip is a zip of [plugin.zip, plugin.deb, plugin.rpm] right?
Is it possible with this action to instead upload them individually?
Just imagining someone wants to download the plugin.zip and this makes them download all 3 and then requires them to unzip just to get the plugin.zip

Reply:
As per my research and I consulted Sujith too, for each package (zip, rpm and deb) we need to create upload-release-asset action separately.Basically repeat the action for each file individually